agent/http.go

```diff
@@ -543,6 +543,19 @@ func (s *HTTPHandlers) wrap(handler endpoint, methods []string) http.HandlerFunc
 		} else {
 			err = s.checkWriteAccess(req)
 
+			// Give the user a hint that they might be doing something wrong if their GET request
+			// has a non-empty body, which can easily happen using curl's --data-urlencode
+			// if specifying request type via "--request GET" rather than "--get".
+			// See https://github.com/hashicorp/consul/issues/11471.
```

Comment:
I think the comment makes the code look more severe/specific than it really is.
How's this generalized one?
```suggestion
			// Warn if the GET request has a non-empty body
```

Reply:
Didn't do exactly what you suggested, but did modify the comment. Let me know if you agree.